lib/frame.rb

```diff
@@ -0,0 +1,59 @@
+# knows if it's a strike, spare or over
+class Frame
+  TOTAL_PINS = 10
+
+  attr_reader :remaining_pins, :bonus, :rolls
+
+  def initialize
+    @bonus = 0
+    @rolls = []
+    @final = false
+    @remaining_pins = TOTAL_PINS
+  end
+
+  def score
+    rolls.sum + bonus
+  end
+
+  def add_roll(pins)
+    raise PinError if pins > remaining_pins
```

Comment:
An error is not raised when the user inputs a negative number or anything that isn't an integer.

Reply:
good spot!